test/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersPartnershipRegisteredAddressControllerSpec.scala

```diff
@@ -38,59 +43,63 @@ class BusinessPartnersPartnershipRegisteredAddressControllerSpec extends Control
   override lazy val views: Views = app.injector.instanceOf[Views]
   val mockAppConfig: FrontendAppConfig = mock[FrontendAppConfig]
   val mockAddressService: AddressService = mock[AddressService]
+  val mockSessionCache: SessionRepository = mock[SessionRepository]
+  val index = 1
   val controller =
     new BusinessPartnersPartnershipRegisteredAddressController(
       commonDependencies,
       views,
       mockActions,
       mockAppConfig,
-      mockAddressService)(mockMcc)
-  val chooseAddressUrl: String = routes.BusinessPartnersChooseAddressController.load(1, NormalMode).url
-  val confirmPartnershipRegAddressUrl: String =
-    routes.BusinessPartnersPartnershipConfirmRegisteredAddressController.load().url
-  val cannotFindAddressUrl: String = routes.BusinessPartnersCannotFindAddressController.load(1, NormalMode).url
-
-  "load" should {
-    "Render the business partner address page" when {
-      "the new business partner pages are enabled" in {
-        setupUserAction()
-        when(mockAppConfig.newBusinessPartnerPagesEnabled).thenReturn(true)
-        val request = FakeRequest()
-        val result = await(csrfAddToken(controller.load())(request))
-
-        status(result) shouldBe OK
-        val page = Jsoup.parse(contentAsString(result))
-        page.title should include("What is the partnership’s registered office address?")
-        page.body.text() should include("What is Test User’s registered office address?")
-        reset(mockActions)
-      }
-    }
+      mockAddressService,
+      mockSessionCache)(mockMcc)
+  val emptyUserAnswers: UserAnswers = UserAnswers(testUserId)
+
+  def createUserAnswers(answers: UkAddressLookup): UserAnswers =
+    UserAnswers(testUserId)
+      .set[UkAddressLookup](UkAddressLookupPage(1), answers)
+      .success
+      .value
+
+  List(NormalMode, CheckMode).foreach { mode =>
+    s"load when in $mode" should {
+      "Render the business partner address page" when {
+        "the new business partner pages are enabled and there are no user answers for the page" in {
+          setupDataRequiredAction(emptyUserAnswers)
+          when(mockAppConfig.newBusinessPartnerPagesEnabled).thenReturn(true)
+
+          val request = FakeRequest()
+          val result = await(csrfAddToken(controller.load(index, mode))(request))
+
+          status(result) shouldBe OK
+          val page = Jsoup.parse(contentAsString(result))
+          page.title should include("What is the partnership’s registered office address?")
+          page.body.text() should include("What is Test User’s registered office address?")
+          reset(mockActions)
+        }
+
+        "the new business partner pages are enabled and there are user answers for the page" in {
```

Comment:
"the new business partner pages are enabled and" - remove as no longer part of this unit test